src/models/parties/parties.js

```diff
@@ -6,40 +6,42 @@ export const DEFAULT_PARTY_COUNT = 5;
 export const MAX_PARTY_COUNT = 10;
 
 export function makeParties() {
-  let names = Set([
-    'asparagus',
-    'corn',
-    'cabbage',
-    'turnip',
-    'radish',
-    'carrot',
-    'onion',
-    'spinach',
-    'mushroom',
-    'eggplant'
-  ]);
+  let names = [
+    {'asparagus': 'rgb(151, 190,17)',
+      id: 1
+    },
+    {'corn': 'rgb(255,255,51)',
+      id: 2
+    },
+    {'cabbage': 'rgb(204,255,153)',
+      id: 3
+    },
+    {'turnip': 'rgb(204,0,204)',
+      id: 4
+    },
+    {'radish': 'rgb(255,0,127)',
+      id: 5
+    },
+    {'carrot':'rgb(255,108,0)',
+      id: 6
+    },
+    {'onion': 'rgb(223,223,172)',
+      id: 7
+    },
+    {'spinach': 'rgb(40,89,12)',
+      id: 8
+    },
+    {'mushroom': 'rgb(255,204,153)',
+      id: 9
+    },
+    {'eggplant':'rgb(102,0,102)',
+      id: 10
+    }
+  ];
+  let randomIdNumber = Math.floor((Math.random() * 10) + 1)
+  let render = names.map(veggie => {
+    
+  }
 
-  let colors = Set([
-    'rgb(18, 147, 154)',
-    'rgb(121, 199, 227)',
-    'rgb(26, 49, 119)',
-    'rgb(255, 152, 51)',
-    'rgb(239, 93, 40)'
-  ]);
-
-  return Map(
-    listOf(
-      () => {
-        const id = names.maxBy(Math.random);
```

Comment:
Really everything in this return statement should stay the same except lines 33-37, that's where you can change things up.

Reply:
So I added this over the older style of objects as you had them. I think in the way I was trying to do this before, I am not sure how to do this any differently because it seems like this is still taking them as separate arrays. Im not sure how to make this operate with name and colors being within one object instead of two. 
`return Map(
    listOf(
      () => {
        const id = potentialParties.name.maxBy(Math.random);
        potentialParties.name = potentialParties.name.remove(id);

        const color = potentialParties.colors.maxBy(Math.random);
        potentialParties.colors = potentialParties.colors.remove(color);

        const value = Map({ id, color });
        return List([id, value]);
      },
      DEFAULT_PARTY_COUNT
    )
  );
}`

Reply:
for whatever reason all of my code isn't going in the code block

Reply:
To do a code block wrap it in "```"

Reply:
To treat it as one thing instead of two, you can get the whole object out of the array instead of just the color.

Reply:
like:
```
const party = potentialParties.maxBy(Math.random);

const id = party.id;
const color = party.color;
```

Reply:
One big caveat, `maxBy` is not a method on regular js arrays, but it is a method on "immutable.js" arrays: https://immutable-js.github.io/immutable-js/docs/#/List/maxBy

So to use this method, you need to wrap `potentialParties` in `List(...)`. Like:
```js
const potentialParties = List([
  {
    id: ...,
    color: ...
  },
  ...
]);
```